contracts/LingoNFT.sol

```diff
@@ -1,188 +1,210 @@
-    function getFirstClassSupply() public view returns (uint256) {
-        return firstClassSupplyCounter;
     }
-    // Function to set the sale start date
+
+    /// @notice Sets the start time for the NFT sale
+    /// @param _startDate The start time as a UNIX timestamp
     function setSaleStartTime(uint256 _startDate) external onlyOwner {
-        //require(_startDate > block.timestamp,"Invalid Start Date");
         saleStartTime = _startDate;
     }
-    // Function to get the sale start date
-    function getSaleStartTime() external view returns (uint256) {
-        return saleStartTime;
-    }
-    // Function to get the current total supply
-    function getTotalSupply() external view returns (uint256) {
-        return _tokenIdCounter;
-    }
-    //FUnction to get the URI of a specific NFT
-    function tokenURI(uint256 tokenId) public view override returns (string memory) {
-        require(_exists(tokenId), "Token does not exist");
-        // Combine the base URI and token-specific URI to get the full metadata URI
-       return _tokenURIs[tokenId];  
-    }
-   // Override _mint function to include uri and tier
-    function _mint(address to, uint256 tokenId, string memory uri, Tier tier) internal {
-        require(!_exists(tokenId), "Token ID already exists");
-        require(to != address(0), "Cannot mint to the zero address");
-        _hasMinted[msg.sender][tier] = true;
-        _tokenIdCounter = _tokenIdCounter + 1;
-        _tokenURIs[tokenId] = uri;
-        _mint(to, tokenId);
-        if (tier == Tier.FIRST_CLASS) {
-            firstClassSupplyCounter += 1;
-        }
-    }
-    // Function to check the current timestamp
-    function getCurrentTimestamp() external view returns (uint256) {
-        return block.timestamp;
-    }
-    // Function to check if the address has already minted a specific NFT Tier
-    function hasMinted(address account, Tier tier) external view returns (bool) {
-        return _hasMinted[account][tier];
-    }
-    // Function to withdraw funds from the contract
+
+    /// @notice Withdraws contract balance to the owner's address
+    /// @dev Only callable by the contract owner
     function withdraw() external onlyOwner {
         uint256 balance = address(this).balance;
-        require(balance > 0, "No funds left to withdraw");
+        if(balance == 0) revert CannotBeZero();
         // Transfer funds to the owner's address
         (bool sent, ) = owner().call{value: balance}("");
-        require(sent, "Transfer failed.");  
+        if(!sent) revert NotSent();
+    }
+
+    /// @notice Airdrops NFTs of a specified tier to multiple addresses
+    /// @param tier The tier of NFTs to airdrop
+    /// @param recipients The addresses to receive the NFTs
+    function airdropNFT(
+        Tier tier,
+        address[] calldata recipients
+    ) external onlyOwner {
+        for (uint256 i = 0; i < recipients.length; i++) {
+            _mint(recipients[i], tier);
```

Comment:
This bypasses the `maxFirstClassSupply` check

Reply:
This is not a bug but a feature

Reply:
Ok just wanted to be sure